From 4f29f1ce4b631d7ba2780abc0c0ea7ba9c8faa04 Mon Sep 17 00:00:00 2001
From: Tamizh Chelvam Raja <quic_tamizhr@quicinc.com>
Date: Wed, 25 Oct 2023 12:39:01 +0530
Subject: [PATCH] wifi: ath12k: Assign bdf and caldb location depends on mem
 mode

bdf and caldb locations are in array format in dts where each
index values are specifc to different mem mode like 1G/512MB/256MB.
So use memmode as index for choosing the bdf and caldb addresses.

ex : bdf_location=<1G address 512MB address 256 address>

Signed-off-by: Tamizh Chelvam Raja <quic_tamizhr@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/qmi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/qmi.c b/drivers/net/wireless/ath/ath12k/qmi.c
index 1f9a4ca..9b24c10 100644
--- a/drivers/net/wireless/ath/ath12k/qmi.c
+++ b/drivers/net/wireless/ath/ath12k/qmi.c
@@ -4280,7 +4280,7 @@ skip_mlo_mem_init:
 						goto out;
 					}
 
-					ab->qmi.target_mem[idx].paddr = caldb_location[0];
+					ab->qmi.target_mem[idx].paddr = caldb_location[ab->hw_params->fw_mem_mode];
 					ab->qmi.target_mem[i].size = caldb_size[0];
 
 					ab->qmi.target_mem[idx].v.ioaddr =
@@ -4381,7 +4381,7 @@ skip_mlo_mem_init:
 				ret = -EINVAL;
 				goto out;
                         }
-			ab->qmi.target_mem[idx].paddr = bdf_location[0];
+			ab->qmi.target_mem[idx].paddr = bdf_location[ab->hw_params->fw_mem_mode];
 			ab->qmi.target_mem[idx].v.ioaddr =
 					ioremap(ab->qmi.target_mem[idx].paddr,
 						ab->qmi.target_mem[i].size);
-- 
2.17.1

